### PR TITLE
Fix valency error in if-statements

### PR DESCRIPTION
**Pull request by charles-cooper**
### What I did
Fix the following error messages:
```
"Valency mismatch between then and else clause: %r %r"
"2-clause if statement must have a zerovalent body: %r"
```
reproducing code:
```python
@private
def retbool() -> bool:
    return True

@public
def repro1():
    if True:
        self.retbool()

@public
def repro2():
    if True:
        self.retbool()
    else:
        pass
```

### How I did it
Remove the checks and instead put each branch into a 'seq'. This should preserve the original intent of the checks, which was to help formal verification by ensuring each branch would have the same effect to the stack size (although previously even with the checks the branches could end up with different stack sizes by adding a semantically empty 'pass' statement to each branch).

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Comment by fubuloubu** [comment]
Can we also change the error messages to say something more explicit than "valency". It may be technically correct but oh boy is it unintelligible to those who don't know what's going on

**Comment by charles-cooper** [comment]
@jacqueswww will do!
@fubuloubu since it's a compiler error I don't think the error message needs to be clarified except maybe to direct the user to file a bug report. Also in this case the error messages have been removed anyway in favor of changing the semantics - so it's moot :smile:

**Comment by fubuloubu** [comment]
Fair enough! As long as those messages don't get shown to the user unless there is an oversight.

File: tests/parser/syntax/test_conditionals.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+@public
+def test_valency_mismatch():
+    if True:
+        self.mkint()
+    else:
+        pass
+    """
+
+    c = get_contract_with_gas_estimation(conditional_return_code)
```
Comment (fubuloubu) [inline tests/parser/syntax/test_conditionals.py]: I think @jacqueswww has a better way to assert that something compiles in `conftest`

**Comment by jacqueswww** [comment]
@charles-cooper flake always bites you ! :package:

**Comment by charles-cooper** [comment]
oops! I gotta put that in my git hooks or something

**Comment by jacqueswww** [comment]
Thanks @charles-cooper !

**Pull request merged**